src/traces/heatmap/plot.js

```diff
@@ -337,7 +337,14 @@ function plotOne(gd, plotinfo, cd) {
 
     if(zsmooth) { // best or fast, works fastest with imageData
         var pxIndex = 0,
+            pixels;
+
+        try {
             pixels = new Uint8Array(imageWidth * imageHeight * 4);
```

Comment:
I thought you might say that... do we have data on how bad this issue is though? This is two `try` blocks per trace (which won't necessarily both fail together), no loops, is that really worrisome? Do you see a robust alternative?